src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -546,14 +545,7 @@ public void Dispose()
                 }
                 finally
                 {
-                    lock (_throttleLock)
-                    {
-                        if (_throttle != null)
-                        {
-                            _throttle.Release();
-                            _throttle = null;
-                        }
-                    }
+                    Interlocked.Exchange(ref _throttle, null)?.Release();
```

Comment:
How can a ThrottledResponse object be used on multiple threads at the same time? It looks like a new ThrottledResponse object is created each time one is needed (inside of GetThrottledResponse). Seems like this code doesn't need to be thread safe.